### PR TITLE
net/cmdline: correctly handle static ip= config

### PR DESCRIPTION
**Pull request by xnox**
It is proto 'none', not 'static' as was mistakenly implemented in
initramfs-tools/cloud-init in the past, yet was never the case in the
klibc ipconfig state file output.

LP: #1861412

### CODE REVIEW

File: cloudinit/net/cmdline.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def _klibc_to_config_entry(content, mac_addrs=None):
     provided here.  There is no good documentation on this unfortunately.
 
     DEVICE=<name> is expected/required and PROTO should indicate if
-    this is 'static' or 'dhcp' or 'dhcp6' (LP: #1621507).
+    this is 'none' (static) or 'dhcp' or 'dhcp6' (LP: #1621507).
```
Comment (OddBloke) [inline cloudinit/net/cmdline.py]: Can we include a reference to https://git.launchpad.net/~usd-import-team/ubuntu/+source/klibc/tree/usr/kinit/ipconfig/README.ipconfig#n32 which seems to document this?

File: tests/unittests/test_net.py [hunk 2]
```diff
@@ -99,7 +99,7 @@
     'type': 'physical',
     'subnets': [{'broadcast': '10.0.0.255', 'control': 'manual',
                  'gateway': '10.0.0.1',
-                 'dns_search': ['foo.com'], 'type': 'static',
+                 'dns_search': ['foo.com'], 'type': 'none',
```
Comment (OddBloke) [inline tests/unittests/test_net.py]: I don't think this is correct; per https://cloudinit.readthedocs.io/en/latest/topics/network-config-format-v1.html#subnet-ip I think this should still be:

```suggestion
                 'dns_search': ['foo.com'], 'type': 'static',
```

Comment (xnox) [reply tests/unittests/test_net.py]: In that case code is buggy somewhere, and I must add the translation from 'none' to 'static' somewhere.

Comment (OddBloke) [reply tests/unittests/test_net.py]: Yep, presumably we're just reusing the value somewhere as they happened to be the same before this change.

**Comment by OddBloke** [comment]
Oracle is the one cloud platform I'm aware of that uses initramfs networking, and they use the "dhcp" protocol, so we shouldn't see any change in behaviour there.

**Pull request merged**